### PR TITLE
WIP: BIP Unification

### PR DESCRIPTION
**Pull request by jmininger**

### CODE REVIEW

File: frontend/src/Frontend.hs [hunk 1]
```diff
@@ -125,7 +122,7 @@ loaderMarkup = divClass "spinner" $ do
 
 newHead :: (Prerender js t m, DomBuilder t m) => (R BackendRoute -> Text) -> m (Event t ())
 newHead routeText = do
-  el "title" $ text "Kadena - Pact Testnet"
+  el "title" $ text "Chainweaver: Wallet & IDE"
```
Comment (mightybyte) [inline frontend/src/Frontend.hs:125]: Nice catch.  I might suggest adding the word "Kadena" in here.  Perhaps "Kadena Chainweaver".

File: frontend/src/Frontend/App.hs [hunk 2]
```diff
@@ -15,7 +15,7 @@
 -- License     :  BSD-style (see the file LICENSE)
 --
 
-module Frontend.ReplGhcjs where
+module Frontend.App where
```
Comment (mightybyte) [inline frontend/src/Frontend/App.hs:18]: Finally!

File: frontend/src/Frontend/Crypto/Ed25519.hs [hunk 3]
```diff
@@ -39,49 +42,162 @@ module Frontend.Crypto.Ed25519
...
+import           Language.Javascript.JSaddle.Types  (ghcjsPure)
+import           Data.Bits ((.|.))
+import           Frontend.Crypto.Signature
+import           Frontend.Crypto.Password
+import Control.Monad.Except
+import Control.Monad.Catch
+
 --
 -- | PrivateKey with a Pact compatible JSON representation.
 newtype PrivateKey = PrivateKey { unPrivateKey :: ByteString }
```
Comment (mightybyte) [inline frontend/src/Frontend/Crypto/Ed25519.hs:74]: It *might* be worth renaming this to more clearly communicate its purpose.  Probably don't want to spend a ton of time on it if it's used all over the place though.

File: frontend/src/Frontend/Editor.hs [hunk 4]
```diff
@@ -136,7 +136,13 @@ makeEditor m cfg = mdo
 
     gen <- liftIO newStdGen
     (quickFixCfg, onCodeFix) <- applyQuickFix (randoms gen) t $ cfg ^. editorCfg_applyQuickFix
+#ifdef  ghcjs_HOST_OS
+    -- Causes a bug where, upon clicking contract tab, everything freezes
+    -- presumably due to some sort of fixpoint bug
+    let codeAnnotations = constDyn []
```
Comment (mightybyte) [inline frontend/src/Frontend/Editor.hs]: Why don't you add a big scary greppable "FIXME", "WARNING", or something similar to highlight this.

File: frontend/src/Frontend/UI/Transfer.hs [hunk 5]
```diff
@@ -1404,6 +1405,9 @@ transferDetails signedCmd = do
           & textAreaElementConfig_initialValue .~ iv
           & initialAttributes %~ (<> "disabled" =: "" <> "style" =: "width: 100%; height: 18em;")
           & textAreaElementConfig_setValue .~ updated preview
+#else
+      let notAvailMsg = el "ul" $ text "This feature is not currently available in the browser"
```
Comment (mightybyte) [inline frontend/src/Frontend/UI/Transfer.hs:1441]: Might be worth a quick check to see if there's a pure Haskell yaml library.

**Review by mightybyte — COMMENTED** [review]
Made some comments.  My review is still in progress, though.  More to come.

**Comment by jmininger** [comment]
CLOSING IN FAVOR OF #691

**Pull request closed without merging**